src/main/scala/tech/cryptonomic/conseil/metadata/MetadataService.scala

```diff
@@ -61,9 +61,31 @@ class MetadataService(
   // fetches entities
   def getEntities(path: NetworkPath): Option[List[Entity]] = entities.get(path)
 
+  // gets current entities
+  def getCurrentEntities(path: NetworkPath): Future[Option[List[Entity]]] = {
+    platformDiscoveryOperations.getEntities.map { allEntities =>
+      if(exists(path))
+        Some(transformation.overrideEntities(path, allEntities, shouldLog = false))
+      else
+        None
+    }
+  }
+
   // fetches table attributes
   def getTableAttributes(path: EntityPath): Option[List[Attribute]] = attributes.get(path)
 
+  // fetches current attributes
+  def getCurrentTableAttributes(path: EntityPath): Future[Option[List[Attribute]]] = {
```

Comment:
Similarly we can check the path right out here, too. This also simplifies the map+flatMap nestings (I usually find out they highlight that some refactoring is possible).
Here we can push a little farther with an OptionT and single mapping
```scala
    if(exists(path)) {
      OptionT(platformDiscoveryOperations.getTableAttributes(path.entity)).map { attributes =>
        transformation.overrideAttributes(path, attributes, shouldLog = false)
      }.value
    } else Future.successful(None)
```
but it's not needed